src/js/components/Pagination/Pagination.js

```diff
@@ -0,0 +1,202 @@
+      return { startIndex, endIndex };
+    };
+
+    const handleClick = (event, nextPage) => {
+      event.persist();
+      const adjustedEvent = event;
+      adjustedEvent.page = nextPage;
+
+      // for controlled use cases, provide user with info on
+      // what range of indices should be displayed given the active page
+      const { startIndex, endIndex } = getItemIndices(nextPage);
+      adjustedEvent.startIndex = startIndex;
+      adjustedEvent.endIndex = endIndex;
+
+      setActivePage(event.page);
+
+      if (onChange) {
+        onChange(event);
+      }
+    };
+
+    const NextIcon = theme.pagination.icons.next;
+    const PreviousIcon = theme.pagination.icons.previous;
+    const iconColor = theme.pagination.icons.color;
+
+    const navProps = {
+      next: {
+        // https://a11y-style-guide.com/style-guide/section-navigation.html#kssref-navigation-pagination
+        'aria-disabled': activePage === totalPages ? 'true' : undefined,
+        disabled: activePage === totalPages,
+        icon: <NextIcon color={iconColor} />,
+        onClick: event => {
+          const nextPage = activePage + 1;
+          handleClick(event, nextPage);
+        },
+        label: undefined,
+      },
+      previous: {
+        'aria-disabled': activePage === 1 ? 'true' : undefined,
+        disabled: activePage === 1,
+        icon: <PreviousIcon color={iconColor} />,
+        onClick: event => {
+          const previousPage = activePage - 1;
+          handleClick(event, previousPage);
+        },
+        label: undefined,
+      },
+    };
+
+    let controls = [
+      'previous',
+      ...beginPages,
+      ...beginFlex, // either "..." or single page to bridge start + middle
+      ...middlePages,
+      ...endFlex, // either "..." or single page to bridge middle + end
+      ...endPages,
+      'next',
+    ];
+
+    /* Set props for each page index. Each page index should display a
+     * clickable index, control, or placeholder (e.g. ellipsis) indicating
+     * more pages are available.
+     */
+    controls = controls.map(control => {
+      return {
+        active: control === activePage,
+        a11yTitle:
+          typeof control === 'number'
+            ? `Go to page ${control}`
+            : `Go to ${control} page`,
+        // https://a11y-style-guide.com/style-guide/section-navigation.html#kssref-navigation-pagination
+        // https://www.w3.org/TR/wai-aria-1.1/#aria-current
+        'aria-current': control === activePage ? 'page' : undefined,
+        control,
+        onClick: event => {
+          handleClick(event, control);
+        },
+        separator: control === 'more-prev' || control === 'more-next',
+        ...navProps[control],
+      };
+    });
+
+    return (
+      <StyledPaginationContainer {...theme.pagination.container} {...rest}>
+        <Nav a11yTitle={a11yTitle || 'Pagination Navigation'} ref={ref}>
+          <Box as="ul" {...theme.pagination.controls}>
```

Comment:
Can we remove this Box and merge it with the Nav above? `<Nav as="ul" .../>`

Reply:
I think we want Pagination to appear as a `nav` in the DOM with the `ul` is its child. I will confirm.

Reply:
Yes, we want nav with ul child. Here's a few links for reference:
https://www.a11ymatters.com/pattern/pagination/
https://a11y-style-guide.com/style-guide/section-navigation.html

Reply:
Thanks for investigating.